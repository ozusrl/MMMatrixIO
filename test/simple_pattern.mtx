%%MatrixMarket matrix coordinate pattern general
5 5 13
1 1
1 2
1 5
2 1
2 3
3 4
4 1
4 2
4 3
4 4
4 5
5 1
5 5
