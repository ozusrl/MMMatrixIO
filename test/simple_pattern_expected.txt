############### MM  ##############
5 5 13
0 0 1
0 1 1
0 4 1
1 0 1
1 2 1
2 3 1
3 0 1
3 1 1
3 2 1
3 3 1
3 4 1
4 0 1
4 4 1
############### LD  ##############
5 5 8
0 0 1
1 0 1
3 0 1
3 1 1
3 2 1
3 3 1
4 0 1
4 4 1
############### UD  ##############
5 5 8
0 0 1
0 1 1
0 4 1
1 2 1
2 3 1
3 3 1
3 4 1
4 4 1
############### COO ##############
int rowIndices[13] = {
0,
0,
0,
1,
1,
2,
3,
3,
3,
3,
3,
4,
4,
};
int colIndices[13] = {
0,
1,
4,
0,
2,
3,
0,
1,
2,
3,
4,
0,
4,
};
double values[13] = {
1,
1,
1,
1,
1,
1,
1,
1,
1,
1,
1,
1,
1,
};
############### CSR ##############
int rowPtr[6] = {
0,
3,
5,
6,
11,
13,
};
int colIndices[13] = {
0,
1,
4,
0,
2,
3,
0,
1,
2,
3,
4,
0,
4,
};
double values[13] = {
1,
1,
1,
1,
1,
1,
1,
1,
1,
1,
1,
1,
1,
};
############### CSC ##############
int rowIndices[13] = {
0,
1,
3,
4,
0,
3,
1,
3,
2,
3,
0,
3,
4,
};
int colPtr[6] = {
0,
4,
6,
8,
10,
13,
};
double values[13] = {
1,
1,
1,
1,
1,
1,
1,
1,
1,
1,
1,
1,
1,
};
